### PR TITLE
raftstore: set sync hint on ingest sst commands

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

After ingest sst, sst files are deleted quickly. As a result, ingest sst command can not be handled again and must be synced.

Set sync hint on ingest sst commands to make sure they are synced properly.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -941,7 +959,7 @@ impl ApplyDelegate {
             );
         }
 
-        if cmd.has_admin_request() {
+        if should_write_to_engine_after(&cmd) {
```
Comment (Little-Wallace) [inline components/raftstore/src/store/fsm/apply.rs]: Why not set it in `handle_ingest_sst`?

File: components/raftstore/src/store/fsm/apply.rs [hunk 2]
```diff
@@ -572,6 +572,24 @@ fn should_write_to_engine(cmd: &RaftCmdRequest, kv_wb_keys: usize) -> bool {
     false
 }
 
+/// Checks if a write is needed to be issued after handling the command.
+fn should_write_to_engine_after(cmd: &RaftCmdRequest) -> bool {
+    if cmd.has_admin_request() {
+        return true;
```
Comment (5kbpers) [inline components/raftstore/src/store/fsm/apply.rs]: Having written to the engine before, do we need to write again?

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: Follow the same style as `should_write_to_engine_before`.

File: components/raftstore/src/store/fsm/apply.rs (hunk 2 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: No, it has not been written to the engine. What has been written are commands before ingest sst. So we need to write it again.

Comment (BusyJay) [reply components/raftstore/src/store/fsm/apply.rs]: I don't think every admin command needs to trigger a write immediately.

File: components/raftstore/src/store/fsm/apply.rs [hunk 3]
```diff
@@ -572,6 +572,24 @@ fn should_write_to_engine(cmd: &RaftCmdRequest, kv_wb_keys: usize) -> bool {
     false
 }
 
+/// Checks if a write is needed to be issued after handling the command.
+fn should_write_to_engine_after(cmd: &RaftCmdRequest) -> bool {
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/apply.rs]: This seems to be completely different from `should_write_to_engine`. It just checks whether need to flush to disk. I suggest to name it as `should_sync_log`.

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (BusyJay) [inline components/raftstore/src/store/fsm/apply.rs]: You can write `apply_ctx.syn_log_hint |= xxx`.

File: components/raftstore/src/store/fsm/apply.rs (hunk 2 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: It does not trigger a write immediately, instead, it just set a sync hint and performs a write at the `fn end(&self)`.

File: components/raftstore/src/store/fsm/apply.rs (hunk 3 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: Sure.

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: Addressed

**Pull request merged**